Sources/ApolloTestSupport/MockGraphQLServer.swift

```diff
@@ -85,14 +90,26 @@ public class MockGraphQLServer {
       return expectation
     }
   }
+
+  public func expect<Operation: GraphQLOperation>(
+    _ operationType: Operation.Type,
+    file: StaticString = #filePath,
+    line: UInt = #line,
+    requestHandler: @escaping (HTTPRequest<Operation>) -> JSONObject
+  ) -> XCTestExpectation {
+    return self.expect(operationType, file: file, line: line) { request -> Result<JSONObject, Error> in
+      let result = requestHandler(request)
+      return .success(result)
+    }
+  }
   
   func serve<Operation>(request: HTTPRequest<Operation>, completionHandler: @escaping (Result<JSONObject, Error>) -> Void) where Operation: GraphQLOperation {
     // Dispatch after a small random delay to spread out concurrent requests and simulate somewhat real-world conditions.
     queue.asyncAfter(deadline: .now() + .milliseconds(Int.random(in: 10...50))) {
       let operationType = type(of: request.operation)
       
       if let expectation = self[operationType] {
-        completionHandler(.success(expectation.handler(request)))
+        completionHandler(expectation.handler(request))
```

Comment:
This seems like an odd change - why does this no longer need to be wrapped in `success`?

Reply:
I think this makes sense if request handlers are changed to return `Result` (so we don't need to wrap the return value again). But I think it would be easier to make result handlers throwing. That way, we could simply do `completionHandler(Result { try expectation.handler(request) })` here.